### PR TITLE
rec: tidy syncres

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

This is a pretty big beast. A few complaints remain. I marked a few functions with the "high-complexity" annotation. Not because I think we should not refactor them, but because I think that should be done in a separate PR. I did handle the more simple cases, though, as they were not very intrusive.
A changed conditions using the clang-tidy suggested substitutions, but given the recent experience with those, beware and double check (I did that as well, but it's easy to oversee something).

It also includes a few cases outside syncres.?? that were either needed (type changes) or just happened to catch my attention.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by omoerbeek** [comment]
Rebased to fix conflicts

File: pdns/recursordist/syncres.cc [hunk 1]
```diff
@@ -909,25 +922,25 @@ bool SyncRes::AuthDomain::operator==(const AuthDomain& rhs) const
 [[nodiscard]] std::string SyncRes::AuthDomain::print(const std::string& indent,
                                                      const std::string& indentLevel) const
 {
-  std::stringstream s;
-  s << indent << "DNSName = " << d_name << std::endl;
-  s << indent << "rdForward = " << d_rdForward << std::endl;
-  s << indent << "Records {" << std::endl;
+  std::stringstream str;
```
Comment (fredmorcos) [inline pdns/recursordist/syncres.cc:925]: `domainInfoString` or `outputString` would be better here.

**Review by fredmorcos — APPROVED** [review]
Just one nitpick but looks fine. Thank you!

File: pdns/recursordist/syncres.cc [hunk 2]
```diff
@@ -1608,15 +1621,15 @@ unsigned int SyncRes::s_max_minimize_count; // default is 10
 /* number of iterations that should only have one label appended */
 unsigned int SyncRes::s_minimize_one_label; // default is 4
 
-static unsigned int qmStepLen(unsigned int labels, unsigned int qnamelen, unsigned int i)
+static unsigned int qmStepLen(unsigned int labels, unsigned int qnamelen, unsigned int iter)
```
Comment (chbruyand) [inline pdns/recursordist/syncres.cc:1624]: `iter` sounds like an iterator to me. Maybe `index` or `idx` ?

File: pdns/recursordist/syncres.cc [hunk 3]
```diff
@@ -4512,10 +4505,11 @@ RCode::rcodes_ SyncRes::updateCacheFromRecords(unsigned int depth, const string&
     }
   }
 
-  for (tcache_t::iterator i = tcache.begin(); i != tcache.end(); ++i) {
+  for (auto i = tcache.begin(); i != tcache.end(); ++i) {
```
Comment (chbruyand) [inline pdns/recursordist/syncres.cc:4508]: maybe rename `i` ?

**Review by chbruyand — APPROVED** [review]
Just a nit. LGTM otherwise.

**Comment by omoerbeek** [comment]
Rebased to fix conflicts

**Pull request merged**